worldedit-core/src/main/java/com/fastasyncworldedit/core/util/MathMan.java

```diff
@@ -200,6 +206,33 @@ public static int tripleSearchCoords(int x, int y, int z) {
                 + ((b4 & 0xFF) << 23));
     }
 
+    /**
+     * Obtain an integer representation of 3 ints, x, y and z. y is represented by the right-most 9 bits and can
+     * be within the range -256 to 255. x and z are represented by 11 bits each and can be within the range
+     * -1024 to 1024
+     *
+     * @param x x value
+     * @param y y value
+     * @param z z value
+     * @return integer representation of x y z
+     */
+    public static int tripleSearchCoordsExtendedY(int x, int y, int z) {
+        if (x > 1023 || x < -1024 || y > 255 || y < -256 || z > 1023 || z < -1024) {
+            throw new IndexOutOfBoundsException(String.format("Check range on x=%s, y=%s and z=%s!", x, y, z));
+        }
+        byte b1 = (byte) (y & 0xff);
```

Comment:
This looks more complex than needed. What's wrong with
```java
return (z << 21) | ((x & 0x7ff) << 10) | (y & 0x3ff);
```
?
If this method is just used for hashing, that's already fine, if the coordinates should be extracted again, it would make sense to add the minimum before the OR to easily deal with negative values.

Reply:
It needs to accept negative values. Coordinates do need to be extracted again. The current way in the PR is better in keeping with how it does done previously, though if there's a method that also works I'm open to it - just PR it as I';ve not got time to think about it for a while...